gpioioctl/basic_test.go

```diff
@@ -0,0 +1,120 @@
+package gpioioctl
+
+// Copyright 2024 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+//
+// Basic tests. More complete test is contained in the
+// periph.io/x/cmd/v3/periph-smoketest/gpiosmoketest
+// folder.
+import (
+	"testing"
+
+	"periph.io/x/conn/v3/driver/driverreg"
+	"periph.io/x/conn/v3/gpio/gpioreg"
+)
+
+var _test_line *GPIOLine
+
+func init() {
+	_, _ = driverreg.Init()
+}
+
+func TestChips(t *testing.T) {
+	if len(Chips) <= 0 {
```

Comment:
Can this test pass on macOS and Windows? Options could be to mock or move this kind of testing to https://github.com/periph/cmd/pull/7.

Reply:
Can we do this with build tags and conditional compilation? Do you specify any --tags args in the go test command?

Reply:
Even so, the test assumes the host will expose chips on linux. This will not be the case on VMs.

Reply:
OK, I've added a mock to make this work regardless if init() finds any gpio chips.